### PR TITLE
[Impeller] Add validation forbidding SamplerAddressMode::kDecal on the OpenGLES backend

### PR DESCRIPTION
**Pull request by bdero**
Fixes uncontrolled crash when kDecal is used (as seen on https://github.com/flutter/engine/pull/43087).

### CODE REVIEW

**Review by jonahwilliams — COMMENTED** [review]
What happens if we attempt to bind a null Sampler?

File: impeller/renderer/backend/gles/sampler_library_gles.cc [hunk 1]
```diff
@@ -17,6 +19,14 @@ SamplerLibraryGLES::~SamplerLibraryGLES() = default;
 // |SamplerLibrary|
 std::shared_ptr<const Sampler> SamplerLibraryGLES::GetSampler(
     SamplerDescriptor descriptor) {
+  if (descriptor.width_address_mode == SamplerAddressMode::kDecal ||
+      descriptor.height_address_mode == SamplerAddressMode::kDecal ||
+      descriptor.depth_address_mode == SamplerAddressMode::kDecal) {
+    VALIDATION_LOG << "SamplerAddressMode::kDecal is not supported by the "
```
Comment (zanderso) [inline impeller/renderer/backend/gles/sampler_library_gles.cc:25]: A TODO linking to a GItHub issue would probably be helpful here.

**Review by zanderso — APPROVED** [review]

File: impeller/renderer/backend/gles/sampler_library_gles.cc (hunk 1 above)
Comment (bdero) [reply impeller/renderer/backend/gles/sampler_library_gles.cc:28]: Done & filed this issue: https://github.com/flutter/flutter/issues/129358

**Comment by bdero** [comment]
> What happens if we attempt to bind a null Sampler?

The `command.BindResource` call will early return and not add the sampler to the command, and then the GLES backend will fail to find the sampler slot. This triggers a chain of early returns that bubble up to an [FML_CHECK](https://github.com/flutter/engine/blob/1225ac5dd6e986b1fe1fecce91408d3eb6db2ed7/impeller/renderer/backend/gles/render_pass_gles.cc#L535) that will fail.

**Pull request merged**